modules/ROOT/pages/domain-based-application-cookbook.adoc

```diff
@@ -0,0 +1,154 @@
+MUnit 2.x allows you to write MUnit test suites for your Mule applications that share their resources with a Mule domain without the need for any extra configurations.
+
+== Base Mule Application
+
+Consider a Mule domain called `domain-config` that defines a global configuration for the HTTP listeners of all the Mule applications under it:
+
+[.ex]
+=====
+[discrete.view]
+===== Studio Visual Editor
+
+image::domain-munit-example.png[]
+
+
+[discrete.view]
+===== XML Editor
+
+.Mule domain
+[source,XML,linenums]
+----
+<domain:mule-domain ...>
+
+<http:listener-config name="HTTP_Listener_config_domain" basePath="/" >
+  <http:listener-connection host="0.0.0.0" port="8081" />
+</http:listener-config>
+
+
+</domain:mule-domain>
+----
+
+.Mule domain pom.xml file
+[source,XML,linenums]
+----
+<groupId>com.mycompany</groupId>
+<artifactId>domain-config</artifactId>
+<version>1.0.0-SNAPSHOT</version>
+<packaging>mule-domain</packaging>
+<name>domain-config</name>
+----
+=====
+
+And a basic Mule application that returns the path being pinged:
+
+[.ex]
+=====
+[discrete.view]
+===== Studio Visual Editor
+
+image::mule-app-domain-listener.png[]
+
+Note that this application does not define a HTTP Listener global configuration because it inherits it from its Mule domain (`HTTP_Listener_config_domain`).
+
+image::mule-app-domain-set-payload.png[]
+
+[discrete.view]
+===== XML Editor
+
+[source,XML,linenums]
+----
+<mule ...>
+
+  <flow name="application-aFlow">
+        <http:listener  config-ref="HTTP_Listener_config_domain" path="/api"/>
+        <set-payload value='#["You reached " ++ message.attributes.listenerPath]'/>
+  </flow>
+
+</mule>
+----
+
+Note that this application does not define an `http:listener-config` global element because it inherits it from its Mule domain (`HTTP_Listener_config_domain`). The application declares the domain dependency in its `pom.xml` file:
+
+[source,XML,linenums]
+----
+<dependencies>
+...
+  <dependency>
+    <groupId>com.mycompany</groupId>
+    <artifactId>domain-config</artifactId>
+    <version>1.0.0</version>
+    <classifier>mule-domain</classifier>
+    <scope>provided</scope>
+  </dependency>
+...
+</dependencies>
+----
+
+=====
+
+== Write The MUnit Test
+
+Because the Mule domain is a dependency of the Mule application, MUnit can start up the Mule domain to pick up the necessary global elements configuration for the application that it's testing.
+
+Assume an MUnit test to validate that the path being pinged is indeed the expected `/api` path:
+
+
+
+[.ex]
+=====
+[discrete.view]
+===== Studio Visual Editor
+
+image::munit-domain-sample.png[]
+
+. Inside the `execution` scope, an HTTP requester sends a `GET` request to your application's endpoint.
+. Inside the `validation` scope, set up an Assert Equals` processor to assert the response from the `execution` scope against your expected response.
+. To be able to compare the response against the expected text you must convert the payload to plain text.
```

Comment:
These are formatted as either steps in a task or a chronological list of events, but they are worded both ways.
Can we make them parallel? 
I think the easiest way to do this is to slightly reword #2 to something like 

"Inside the `validation` scope, you decide to set up an Assert Equals` processor to assert the response from the `execution` scope against your expected response."

Something that does not use imperative voice (do this).

Also, "To be able to compare the response against the expected text you must..."  >  "To compare the response against the expected text, you must..."

Reply:
These are not steps. These are callouts from the code callouts, or from the callouts in the screenshots.
Ideally, we should use <1>, instead of numbered lists, but we need to fix the style of the site for those to display properly.